components/script/dom/htmlmediaelement.rs

```diff
@@ -670,13 +681,80 @@ impl HTMLMediaElement {
         }
     }
 
+    fn fetch_request(&self, offset: Option<u64>) {
+        if self.resource_url.borrow().is_none() {
+            eprintln!("Missing request url");
+            self.queue_dedicated_media_source_failure_steps();
+            return;
+        }
+
+        // FIXME(nox): Handle CORS setting from crossorigin attribute.
+        let document = document_from_node(self);
+        let destination = match self.media_type_id() {
+            HTMLMediaElementTypeId::HTMLAudioElement => Destination::Audio,
+            HTMLMediaElementTypeId::HTMLVideoElement => Destination::Video,
+        };
+        let mut headers = Headers::new();
+        headers.set(HyperRange::Bytes(vec![ByteRangeSpec::AllFrom(
+            offset.unwrap_or(0),
+        )]));
```

Comment:
does it worth to add this header if offset is zero?

Reply:
For simplicity, yes :). A server not supporting byte range requests will simply ignore the header.